test/common/router/config_impl_test.cc

```diff
@@ -2635,6 +2637,197 @@ TEST(CustomRequestHeadersTest, CustomHeaderWrongFormat) {
       "%CLIENT_IP");
 }
 
+TEST(MetadataMatchCriteriaImpl, Create) {
+  auto v1 = ProtobufWkt::Value();
+  v1.set_string_value("v1");
+  auto v2 = ProtobufWkt::Value();
+  v2.set_number_value(2.0);
+  auto v3 = ProtobufWkt::Value();
+  v3.set_bool_value(true);
+
+  auto metadata_struct = ProtobufWkt::Struct();
+  auto mutable_fields = metadata_struct.mutable_fields();
+  mutable_fields->insert({"a", v1});
+  mutable_fields->insert({"b", v2});
+  mutable_fields->insert({"c", v3});
+
+  auto matches = MetadataMatchCriteriaImpl(metadata_struct);
+
+  EXPECT_EQ(matches.metadataMatchCriteria().size(), 3);
+  auto it = matches.metadataMatchCriteria().begin();
+  EXPECT_EQ((*it)->name(), std::string("a"));
+  EXPECT_EQ((*it)->value().value().string_value(), std::string("v1"));
+  it++;
+
+  EXPECT_EQ((*it)->name(), std::string("b"));
```

Comment:
Do you really need `std::string` here? I would have guessed there would be implicit constructor invocation. This is actually a bit dangerous for the Google import, since our proto string type is not `std::string`, we rely in various places (such as what's being done here) for the implicit constructor to get us to our internal string type.

Reply:
Hm. I recall getting a compiler error that required this to fix it, but I can't reproduce it. I'll take them out.